test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -892,6 +892,26 @@ TEST_F(HttpConnectionManagerConfigTest, ServerPassThrough) {
             config.serverHeaderTransformation());
 }
 
+TEST_F(HttpConnectionManagerConfigTest, SchemeOverwrite) {
+  const std::string yaml_string = R"EOF(
+  stat_prefix: ingress_http
+  scheme_header_transformation:
+    scheme_to_overwrite: http
+  route_config:
+    name: local_route
+  http_filters:
+  - name: envoy.filters.http.router
+  )EOF";
+
+  EXPECT_CALL(context_.runtime_loader_.snapshot_, featureEnabled(_, An<uint64_t>()))
+      .WillRepeatedly(Invoke(&context_.runtime_loader_.snapshot_,
+                             &Runtime::MockSnapshot::featureEnabledDefault));
+  HttpConnectionManagerConfig config(parseHttpConnectionManagerFromYaml(yaml_string), context_,
+                                     date_provider_, route_config_provider_manager_,
+                                     scoped_routes_config_provider_manager_, http_tracer_manager_,
+                                     filter_config_provider_manager_);
```

Comment:
This doesn't really test anything. Can you query the value from the config at least?

Reply:
whoops - lost the actual test on merge :-P